document for E0502 insufficiently general

### Location

Location is E0502.md, the explanation for Error code E0502

https://github.com/rust-lang/rust/blob/master/compiler/rustc_error_codes/src/error_codes/E0502.md

online version: https://doc.rust-lang.org/error_codes/E0502.html

### Summary

The doc explains E0502 as follows:

> A variable already borrowed as **immutable** was borrowed as **mutable**.

However, E0502 is also emitted when a variable already borrowed as **mutable** was borrowed as **immutable**. E.g. see line 1 and 13 in [this file](https://github.com/rust-lang/rust/blob/f8e566053207b4ecbcbc7a7d6ded82c43061e3da/tests/ui/borrowck/borrow-raw-address-of-borrowed.stderr#L1)

```
1     error[E0502]: cannot borrow `x` as mutable because it is also borrowed as immutable
...
13    error[E0502]: cannot borrow `x` as immutable because it is also borrowed as mutable
```

So it seems a more general explanation might be advisable that covers (at least) both types of cases.

--- a/compiler/rustc_error_codes/src/error_codes/E0502.md
+++ b/compiler/rustc_error_codes/src/error_codes/E0502.md
@@ -1,4 +1,5 @@
-A variable already borrowed as immutable was borrowed as mutable.
+A variable already borrowed with a certain mutability (either mutable or
+immutable) was borrowed again with a different mutability.
 
 Erroneous code example:
 
@@ -13,7 +14,7 @@ fn foo(a: &mut i32) {
 ```
 
 To fix this error, ensure that you don't have any other references to the
-variable before trying to access it mutably:
+variable before trying to access it with a different mutability:
 
 ```
 fn bar(x: &mut i32) {}

